src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -670,7 +653,61 @@ private static FunctionCompletedMessage CreateCompletedMessage(FunctionStartedMe
                 ParameterLogBlob = startedMessage.ParameterLogBlob
             };
         }
-                
+
+        // Called very early when function is started; before arguments are bound. 
+        private Task NotifyPreBindAsync(FunctionInstanceLogEntry fastItem)
+        {
+            fastItem.LiveTimer = Stopwatch.StartNew();
+
+            if (_functionEventCollector == null)
+            {
+                return Task.CompletedTask;
+            }            
+
+            // Log pre-bind event. 
+            return _functionEventCollector.AddAsync(fastItem);
```

Comment:
The problem with making them virtual is all these classes are internal. So it would be exposing significant new surface area for script to get at them.  

---
In reply to: [126788620](https://github.com/Azure/azure-webjobs-sdk/pull/1218#discussion_r126788620) [](ancestors = 126788620)